### DISCUSSION TITLE
Development of ESPx and RS-485 hardware solutions

### DISCUSSION THREAD

**Discussion by TripitakaBC** [esphome-econet/esphome-econet#124] [General]
This seems to be a great replacement for the cloud-based Rheem API integration but is held back from greater ambitions due to the lack of availability of the m5stack hardware.

I'm willing to put together a hardware tutorial but some changes to the code are likely to be required. I've done some work with modbus and RS-485 protocols in projects on here so I'm not completely new but I don't have the bandwidth to rewrite the project either.

The initial hurdle is that the code is written for the m5stack hardware but alternatives are necessary for the MAX485 (with flow control required in the code) and the arguably better (simpler) RS-485 with automatic flow control.

With most of the code being pulled from Git and substitutions, it limits the amount of customisation (yeah, Apple, I'm side-eying you here) and also development without forks. I've used the code 'as-is' but I'm all out of the RS-485 automatic flow-control units so I used a MAX485 I had on hand but with the same results I've had before...bus silence. Seems these MAX485 require a flow control code but assembling from the Git code to introduce it results in validation errors elsewhere.

Just figured I'd throw this out there and I'll try again later in the week when I get some RS-485 auto boards to play with.

**Comment by tronikos** [comment]
No changes are needed for using an RS-485 with automatic flow control. I'm using it successfully with ESP8266 and HW-0519. To use MAX485 a few code changes are required though. Search for `flow_control_pin` in https://github.com/syssi/esphome-solax-x1-mini for what needs to be changed.

**Reply by TripitakaBC** [reply]
Yeah, I figured that would be the case but I never did manage to get the MAX485 working on Esphome with other projects so for now, I'll just wait for the auto's to arrive.

I think that there is a lot I can do coming at this project with fresh eyes; I was going round in circles trying to figure out why only one end of the RJ11 cable gets cut off, for example. I eventually figured out that I'm not wiring this into the wall controller and somewhere, there must be an RJ11 socket on my HVAC. Sure enough, once I moved the stuff piled up in from of it and pulled the covers, there it was. Also, the RJ11 schema is correct for one end of the cable; the other end is reversed. Guess who had a couple of RJ11 cables from other projects that had the 'wrong' end on them? Yeah, this guy.... :-D

**Reply by Daniel-dev22** [reply]
> Yeah, I figured that would be the case but I never did manage to get the MAX485 working on Esphome with other projects so for now, I'll just wait for the auto's to arrive.
> 
> I think that there is a lot I can do coming at this project with fresh eyes; I was going round in circles trying to figure out why only one end of the RJ11 cable gets cut off, for example. I eventually figured out that I'm not wiring this into the wall controller and somewhere, there must be an RJ11 socket on my HVAC. Sure enough, once I moved the stuff piled up in from of it and pulled the covers, there it was. Also, the RJ11 schema is correct for one end of the cable; the other end is reversed. Guess who had a couple of RJ11 cables from other projects that had the 'wrong' end on them? Yeah, this guy.... :-D

What kind of HVAC system do you have? What do you mean by wall controller? An external wifi module?


You also don't need to use the yaml files in the repo you can make your own. I don't use them and depending on your HVAC system I might finally take the time to get a pr up for a ton of sensors to be added to the HVAC yaml as there hasn't been any other active HVAC user.

**Reply by TripitakaBC** [reply]
Prestige, I think. Model is RP2036BJVCA with the Econet wifi wall thermostat.

**Reply by TripitakaBC** [reply]
I figure if this is made easy for people to build, more will use it. Simplified wiring diagrams for ESP8266 and ESP32 with a link to files for a 3D printed case for each would really get it rolling. I have some case designs that I could utilise for that already.

**Reply by Daniel-dev22** [reply]
> Prestige, I think. Model is RP2036BJVCA with the Econet wifi wall thermostat.

Awesome I have the same type RP20. I have an external wifi module though so I didn't have to open up the system to get to the HVAC board to connect the esp32 to.

Will look to get a pr up to upload all the diagnostic sensors we can get from the heat pump. There's a lot of diagnostic data from the airhandler I can't figure out because the response isn't the same as anything else on the protocol. When you get the component up maybe you might have some luck figuring out what positions of the data tie to which parts I only figured out the airhandler CFM. RPM sometimes is off which is odd other times spot-on.

Then the airhandler run time diagnostics I have that figured out but not register temperature, return temperature, static pressure and probably a few others I'm missing.

**Comment by tronikos** [comment]
Filed https://github.com/esphome-econet/esphome-econet/issues/125 If you have MAX485, contributions are welcome!

**Reply by tronikos** [reply]
https://github.com/esphome-econet/esphome-econet/pull/126 adds support for flow control pin.

**Reply by tronikos** [reply]
@TripitakaBC latest version supports MAX485 See https://github.com/esphome-econet/esphome-econet/wiki/Schematics

**Reply by kpfleming** [reply]
Also works with the ADM3483 as used on the Olimex MOD-RS485, although it is necessary to solder a wire jumper on the module to tie together the DE and RE pins.

**Reply by dcj** [reply]
@kpfleming 
> Also works with the ADM3483 as used on the Olimex MOD-RS485, although it is necessary to solder a wire jumper on the module to tie together the DE and RE pins.

This is GTK, thanks!
Do you have any advice as to how best to accomplish this jumper?  Can this be done on the Olimex ESP32 board (ESP32-POE-ISO) somehow, or must it be done on the MOD-RS485 itself?

**Reply by kpfleming** [reply]
It has to be done on the MOD-RS485, although it would also be possible to add a feature to the esphome-econet component to control two flow-control pins instead of one.
